### ISSUE TITLE
Configurable reporting interval for default custom metrics and performance counters

### ISSUE THREAD

**Issue by LumirH** [microsoft/ApplicationInsights-Java#1493]
I can configure a hearthbeat or micrometer interval:

"heartbeat": {
    "intervalSeconds": 360
 }

micrometer{
  reportingInternalSeconds:360
}

I would like to also have option to configure in similar way reporting interval for custom metrics and default performance counters. (i.e.  https://github.com/microsoft/ApplicationInsights-Java/wiki/Default-Metrics-Captured-by-Java-3.0-Agent#default-metrics-captured-as-custom-metrics )

These metrics are by default sent every 60 seconds.

**Comment by ashwajce** [comment]
+1

**Comment by trask** [comment]
hey @LumirH @ashwajce (and everyone reading this issue)

the micrometer `reportingInternalSeconds` was an undocumented feature that we didn't hide well enough (and cleverly found!)

we added official (preview) support for this feature under a new setting in [3.0.3-BETA](https://github.com/microsoft/ApplicationInsights-Java/releases/tag/3.0.3-BETA) (and removed the old setting):

```
{
  "preview": {
    "metricIntervalSeconds": 60
  }
}
```


The new setting applies to all of these metrics:

* Default performance counters, e.g. CPU and Memory
* Default custom metrics, e.g. Garbage collection timing
* Configured JMX metrics
* Micrometer metrics

I just merged a doc PR with this new setting, so it should show up on https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-config within the next 24 hours.
